src/AutoRest.CSharp.V3/Generation/Writers/RestClientWriter.cs

```diff
@@ -113,39 +113,58 @@ private void WriteRequestCreation(CodeWriter writer, RestClientMethod operation)
 
                 writer.Line($"var {message:D} = _pipeline.CreateMessage();");
                 writer.Line($"var {request:D} = {message}.Request;");
-                var method = operation.Request.HttpMethod;
+                var method = clientMethod.Request.HttpMethod;
                 writer.Line($"{request}.Method = {typeof(RequestMethod)}.{method.ToRequestMethodName()};");
 
-                //TODO: Add logic to escape the strings when specified, using Uri.EscapeDataString(value);
-                //TODO: Need proper logic to convert the values to strings. Right now, everything is just using default ToString().
-                //TODO: Need logic to trim duplicate slashes (/) so when combined, you don't end  up with multiple // together
-
                 writer.Line($"var {uri:D} = new RawRequestUriBuilder();");
-                foreach (var segment in operation.Request.HostSegments)
+                Parameter? nextLinkParameter = parameters.FirstOrDefault(p => p.Name == "nextLink");
+                bool hasArtificialNextLink = nextLinkParameter?.IsArtificial == true;
+                PathSegment? nextLinkSegment = clientMethod.Request.PathSegments.FirstOrDefault(s => !s.Value.IsConstant && s.Value.Reference.Name == "nextLink");
+                // Artificial nextLink needs additional logic for relative versus absolute links
+                if (hasArtificialNextLink)
                 {
-                    WriteUriFragment(writer, uri, segment);
+                    using (writer.Scope($"if ({nextLinkParameter!.Name:D}.StartsWith({typeof(Uri)}.UriSchemeHttp, {typeof(StringComparison)}.InvariantCultureIgnoreCase))"))
```

Comment:
Can we put this logic into AppendRawNextLink?

Reply:
Unfortunately, we cannot. If you look at what was changed in the generated output, if this if-statement is false, we write host + non-parsed nextLink (as a parameter). If this fix statement is true, we only write nextLink as the entire url (via AppendRaw). Essentially, this is wrapping the dynamic amount of host segments.

Reply:
We can reset the URI as part of `AppendRawNextLink`

Reply:
Hmm. I'll experiment with this. It is kinda bad that we'd write the host segments and then just remove them. But, it does simplify the `CreateRequest` writer.

Reply:
Done. Now, `AppendRawNextLink` is used for artificial nextLink values and contains this if-statement logic.